app/src/main/java/atlant/moviesapp/fragments/AiringTvFragment.java

```diff
@@ -59,20 +65,60 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         GridLayoutManager mLayoutManager = new GridLayoutManager(getActivity().getApplicationContext(), 2);
         recyclerView.setLayoutManager(mLayoutManager);
         adapter = new TVListAdapter(series, getActivity().getApplicationContext());
+        adapter.setItemClick(new OnItemClick() {
+            @Override
+            public void onfavouriteClicked(int position) {
+                if (ApplicationState.isLoggedIn()) {
+                    TvShow m = series.get(position);
+                    if (ApplicationState.getUser().getFavouriteSeries().contains(m.getId())) {
+                        ApplicationState.getUser().removeFavoriteShow(m.getId());
+                        BodyFavourite bodyFavourite = new BodyFavourite(getString(R.string.tv), m.getId(), false);
+                        presenter.postFavorite(m.getId(), ApplicationState.getUser().getSessionId(), bodyFavourite);
+                        Toast.makeText(getActivity().getApplicationContext(), getString(R.string.removedFavorite), Toast.LENGTH_SHORT).show();
+
+                    } else {
+                        ApplicationState.getUser().addFavouriteShow(m.getId());
+                        BodyFavourite bodyFavourite = new BodyFavourite(getString(R.string.tv), m.getId(), true);
+                        presenter.postFavorite(m.getId(), ApplicationState.getUser().getSessionId(), bodyFavourite);
+                        Toast.makeText(getActivity().getApplicationContext(), getString(R.string.addedFavorite), Toast.LENGTH_SHORT).show();
+
+                    }
+                } else {
+                    showLoginError();
+                }
+            }
 
-        recyclerView.addOnItemTouchListener(new TVListAdapter.RecyclerTouchListener(getActivity(), recyclerView, new TVListAdapter.ClickListener() {
             @Override
-            public void onClick(View view, int position) {
+            public void onwatchlistClicked(int position) {
+                TvShow m = series.get(position);
+                if (ApplicationState.isLoggedIn()) {
+                    if (ApplicationState.getUser().getWatchListSeries().contains(m.getId())) {
+                        ApplicationState.getUser().removeWatchlistShow(m.getId());
+                        BodyWatchlist bodyFavourite = new BodyWatchlist(getString(R.string.tv), m.getId(), false);
```

Comment:
This one comment will apply for all actions that are either adding, removing movies and series either from the watchlist or the favourites list. 
Could we move the process of generating/instantiating the BodyWatchList object, or any object that is posted to the post method -> to the presenter.